### DISCUSSION TITLE
misconfiguration/proxy templates timeout

### DISCUSSION THREAD

**Discussion by savushkin-yauheni** [projectdiscovery/nuclei-templates#3646] [Q&A]
Hi team

I'm trying to test `misconfiguration/proxy` templates, e.x. 
https://github.com/projectdiscovery/nuclei-templates/blob/master/misconfiguration/proxy/metadata-google.yaml

So for any other template:
```
time (nuclei -t technologies/cockpit-detect.yaml  -rl 500 -bs 200 -l first.txt)

34.840 total

```

For proxy template:

```
time (nuclei -t misconfiguration/proxy/metadata-google.yaml  -rl 500 -bs 200 -l second.txt)

!!!! 1:25.89 total !!!
```

in first.txt 500 web hosts.
in second.txt !!!30!!! web hosts

Maybe for unsafe templates we should use some special options ? Or why is it so slow ?

Thanks!

**Comment by ehsandeep** [comment]
@savushkin-yauheni I'm not sure what you mean by slow, if the server does not respond to HTTP request which it will in case of proxy templates as the server respone is only expected in case of vulernable host, otherwise it will consume the default timeout period for all the requests nuclei makes unlike in other templates where server responds for valid HTTP request that gets completed quickly.

You can also control the timeout period with `timeout` flag while running these templates.

let me know if i misunderstood your question.

**Reply by savushkin-yauheni** [reply]
Hi @ehsandeep 
Thanks for reply.

But, if default timeout is 5 seconds and `bs` option is `200`, so for `30` web hosts what is a  max execution time ? 
I thought that around 5 seconds +  nuclei logic time. But not `85` seconds. Or am i missing something ?

**Reply by savushkin-yauheni** [reply]
@ehsandeep 
I tested it again:
30 hosts  ±100 seconds
60 hosts ±450 seconds !!!



So, i think there is a problem.


You can retest:

[result.txt](https://github.com/projectdiscovery/nuclei-templates/files/7977933/result.txt)

```
time (nuclei -t misconfiguration/proxy/metadata-google.yaml  -rl 500 -bs 200 -l result.txt  )
```
